### ISSUE TITLE
QSelect throws error when using a function for `option-value` and initial model `null`

### ISSUE THREAD

**Issue by Evertvdw** [quasarframework/quasar#11754]
**Describe the bug**
An error is thrown when using a function as `option-value` together with a starting model value of `null`. If the initial model is an empty string there is no error and everything works as expected.

**Codepen/jsFiddle/Codesandbox (required or your issue may be closed)**
https://codepen.io/Evertvdw/pen/BawwNMy?editors=1010

**To Reproduce**
1. Open select
2. Select an options
3. See error in console

**Expected behavior**
No error, having an initial model of null should be possible.

**Platform (please complete the following information):**
Quasar Version: 2.3.4

**Additional context**
The error in the code is on this line:
https://github.com/quasarframework/quasar/blob/041f33ff13e466ab7528f0dc86d17bd321fccf16/ui/src/components/select/QSelect.js#L190
Here a check is done for `!== null`, and if the modelValue is null it will return an empty array. Of this empty array the first element is accessed and passed into the `optionValue` function here: 
https://github.com/quasarframework/quasar/blob/041f33ff13e466ab7528f0dc86d17bd321fccf16/ui/src/components/select/QSelect.js#L514

**Comment by pdanpdan** [comment]
the function should check what value it gets and return the proper response
https://codepen.io/pdanpdan/pen/MWEEaJr?editors=1010

I agree the example in the docs (API card) is not good :)
```
:option-value="(item) => item === null ? null : item.modelNumber"
```

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.4.2.
Not a docs one. `options-value` should not care about the value being `null`, so leave it as is and upgrade to v2.4.2 (to be released shortly).
